continuous-integration/linux/build.sh

```diff
@@ -4,6 +4,12 @@ SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 ROOT_DIR=$(realpath "$SCRIPT_DIR/../..")
 BUILD_ROOT_DIR="$ROOT_DIR/build"
 
+#TODO 16.04 default pcl is too old. See issue #43
+source /etc/os-release || exit $?
+if [[ "$VERSION_ID" == "16.04" ]]; then
+    OS_SPECIFIC_OPTIONS="-DUSE_PCL=OFF"
```

Comment:
Looking at `azure-pipelines.yml` it looks like 16.04 is first class citizen.

I wonder if `build.sh` should be kept platform independent, then we have two options:
 1. This and the `if` below should be handled as cli options to build.sh so that we have full transparency of the quirks in `azure-pipelines.yml`.
 2. Fix `ubuntu-16.04/setup.sh` so that build.sh works. Then we also document how to fully set up a working environment.
 3. Combinations of the above

I prefer option (2.). Do all the fixes/quirks in `setup.sh` leave `build.sh` unchanged and keep `azure-pipelines.yml` clean.

Reply:
16.04 should absolutely be a first class citizen. This `if` goes away when I do #43, which is very high pri.

After #43 is fixed, "fix `setup.sh` so that `build.sh` works" would mean installing a non system-default version of clang. Since we claim to support 16.04, I think it is important to test on the default clang on 16.04. We have an internal issue in Zivid to fix Clang 3 support, so the second `if` will hopefully also go away in a few months. In the mean time, I think option 1 is best.

Reply:
@nedrebo I've pushed a suggestion now, what do you think?